File: CompilationTest/ExplicitGenericTypeTest/results/ExplicitGenericTypeTest.cpp
#include <OneLang-Core-v0.1/one.hpp>
#include <iostream>
#include <map>

class TestClass {
  public:
    void testMethod() {
        auto result = std::make_shared<std::vector<std::string>>(std::initializer_list<std::string>{  });
        auto map = make_shared_map<std::string, int>({
          { "x", 5 }
        });
        auto keys = OneMapHelper::keys(map);
        std::cout << result << std::endl;
        std::cout << keys << std::endl;
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}

File: CompilationTest/ExplicitGenericTypeTest/results/ExplicitGenericTypeTest.rb
class TestClass
  def test_method()
      result = ["y"]
      map = {
        "x" => 5,
      }
      keys = map.keys
      puts result[0]
      puts keys[0]
  end
end

begin
    TestClass.new().test_method()
rescue Exception => err
    print "Exception: #{err}"
end